src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTask.cs

```diff
@@ -79,9 +84,12 @@ public override bool Execute()
         {
             var request = GetRequest();
             var logic = PackTaskLogic;
-
+            PackageBuilder packageBuilder = null;
             var packArgs = logic.GetPackArgs(request);
-            var packageBuilder = logic.GetPackageBuilder(request);
+            if (string.IsNullOrEmpty(request.NuspecFile))
```

Comment:
It's not quite trivial from just looking at the code. Could you add a comment explaining the logic here?
Also, looking forward to tests covering both code paths.